paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -357,25 +388,22 @@ public MergeFunction<KeyValue> create(@Nullable int[][] projection) {
                                                 newRowType, newSequenceFields));
                             }
                         });
-                for (int i = 0; i < projects.length; i++) {
```

Comment:
Why removing this code?

Reply:
Oh it is a mistake when I attempted to remove the codes related to primaryKeyMask. I've added it back.